dbt/task/generate.py

```diff
@@ -127,6 +128,15 @@ def _get_manifest(self):
         return manifest
 
     def run(self):
+        compile_results = None
+        if self.args.compile:
+            compile_results = super(GenerateTask, self).run()
+            if any(r.errored for r in compile_results):
+                dbt.ui.printer.print_timestamped_line(
+                    'compile failed, cannot generate docs'
+                )
+                return
```

Comment:
did you try just returning the compile_results here? I imagine that `handle_and_check` would print out the compilation errors

Reply:
I think the way I overrode it I'll have to return `{'compile_results': compile_results}`, but that's a good idea.